src/Function.cpp

```diff
@@ -441,6 +489,9 @@ Function::make_random_signature(const CGContext& cg_context, const Type* type, c
 
 	if(CGOptions::func_attr_aligned() && rnd_flipcoin(FuncAttrAligned))
 		f->func_attr_aligned = true;
+
+	if(CGOptions::func_attr_flag())
```

Comment:
Handle that once in `Function::GenerateAttributes`.

Reply:
> Handle that once in `Function::GenerateAttributes`.

I did not get it "Handle that once in `Function::GenerateAttributes`". Do you mean shifting `common_func_attributes` to `Function::GenerateAttributes`

Reply:
Put the condition `if(CGOptions::func_attr_flag())` into `Function::GenerateAttributes` because you'll potentially handle multiple options in the future.

Reply:
Got it. Should I push suggested changes now.

Reply:
> Got it. Should I push suggested changes now.

Yes please.

Reply:
I will also release dynamic memory holded by `vector<FunctionAttribute*>` in `Function::doFinalization` method which release all the dyanamic memory at the end.